src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -0,0 +1,255 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+
+import antlr.CommonHiddenStreamToken;
+import antlr.RecognitionException;
+import antlr.Token;
+import antlr.TokenStreamException;
+import antlr.TokenStreamHiddenTokenFilter;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaLexer;
+import com.puppycrawl.tools.checkstyle.grammars.GeneratedJavaRecognizer;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Helper methods to parse java source files.
+ *
+ * @author Oliver Burn
+ * @author Pavel Bludov
+ */
+public final class JavaParser {
+
+    /**
+     * Enum to be used for test if comments should be used.
+     */
+    public enum Options {
+
+        /**
+         * Comments nodes should be processed.
+         */
+        WITH_COMMENTS,
+
+        /**
+         * Comments nodes should be ignored.
+         */
+        WITHOUT_COMMENTS
+
+    }
+
+    /** Stop instances being created. **/
+    private JavaParser() {
+    }
+
+    /**
+     * Static helper method to parses a Java source file.
+     * @param contents contains the contents of the file
+     * @return the root of the AST
+     * @throws CheckstyleException if the contents is not a valid Java source
+     */
+    public static DetailAST parse(FileContents contents)
+            throws CheckstyleException {
+        final String fullText = contents.getText().getFullText().toString();
+        final Reader reader = new StringReader(fullText);
+        final GeneratedJavaLexer lexer = new GeneratedJavaLexer(reader);
+        lexer.setCommentListener(contents);
+        lexer.setTokenObjectClass("antlr.CommonHiddenStreamToken");
+
+        final TokenStreamHiddenTokenFilter filter = new TokenStreamHiddenTokenFilter(lexer);
+        filter.hide(TokenTypes.SINGLE_LINE_COMMENT);
+        filter.hide(TokenTypes.BLOCK_COMMENT_BEGIN);
+
+        final GeneratedJavaRecognizer parser = new GeneratedJavaRecognizer(filter);
+        parser.setFilename(contents.getFileName());
+        parser.setASTNodeClass(DetailAST.class.getName());
+        try {
+            parser.compilationUnit();
+        }
+        catch (RecognitionException | TokenStreamException ex) {
+            final String exceptionMsg = String.format(Locale.ROOT,
+                "%s occurred while parsing file %s.",
+                ex.getClass().getSimpleName(), contents.getFileName());
+            throw new CheckstyleException(exceptionMsg, ex);
```

Comment:
When I was talking about this at https://github.com/checkstyle/checkstyle/pull/5437#discussion_r160051077 I didn't mean to throw the same exception as TreeWalker. Please wait for @romani to respond to discussion.
I don't think there is a check that will be against `throw new CheckstyleException(ex);`.

Reply:
We could wrap this around `ParseException`.

Reply:
done.